### DISCUSSION TITLE
Change the color of the Button when it is hovered

### DISCUSSION THREAD

**Discussion by THEKetsu** [emilk/egui#3356] [General]
``` 
let color_button = if is_hover(){
    Color32::WHITE
    } else {
    Color32::RED
    }
let button = Button::new("Hover Me").fill(color_button);
    
    if ui.add(button).hovered(){
        ...
    } 
```
Do you have a simple method to be able to change button color in a simple way ?

**Comment by YgorSouza** [comment]
`ui.style_mut().visuals.widgets.hovered.weak_bg_fill = color;`

The change is valid within the current UI scope.

Example:

```rs
use eframe::{egui, epaint::Color32};

fn main() -> Result<(), eframe::Error> {
    eframe::run_native(
        "My egui App",
        Default::default(),
        Box::new(|_cc| Box::<MyApp>::default()),
    )
}

#[derive(Default)]
struct MyApp {}

impl eframe::App for MyApp {
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
        egui::CentralPanel::default().show(ctx, |ui| {
            ui.scope(|ui| {
                ui.style_mut().visuals.widgets.hovered.weak_bg_fill = Color32::RED;
                ui.button("Red");
                ui.button("Still red");
                ui.style_mut().visuals.widgets.hovered.weak_bg_fill = Color32::BLUE;
                ui.button("Blue");
            });
            ui.button("Default");
        });
    }
}
```
[Screencast from 2023-09-21 19-59-21.webm](https://github.com/emilk/egui/assets/43298013/1277bfc0-d009-4e65-8e3a-5787260da582)

Other methods are being planned, like #3284, but currently this is the way to do it.

**Reply by THEKetsu** [reply]
Thank you very much for the tips ^^

**Reply by sukhmel** [reply]
~~this method seems to no longer work in 0.33~~

your example code works, but it doesn't work in my elaborate case, I'll try to debug it then

Upd: that was because I was filling a button, thus overriding the hover effects. What I don't quite understand is why frame and text hover effects still exist, but that's another story, I think

Upd2: I can't quite understand how should I set colour when hovered manually, as `fill` is called before adding to the UI, and `hovered` is only known after adding. I'm sure I'm missing something though
